projects/observability/src/shared/dashboard/data/graphql/trace-calls-percentage/trace-calls-percentage-data-source.model.ts

```diff
@@ -0,0 +1,71 @@
+import { forkJoinSafeEmpty, getPercentage } from '@hypertrace/common';
+import {
+  GraphQlDataSourceModel,
+  MetricAggregation,
+  MetricAggregationType,
+  MetricHealth
+} from '@hypertrace/distributed-tracing';
+import { Model } from '@hypertrace/hyperdash';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { ExploreSpecification } from '../../../../graphql/model/schema/specifications/explore-specification';
+import { ExploreSpecificationBuilder } from '../../../../graphql/request/builders/specification/explore/explore-specification-builder';
+import {
+  ExploreGraphQlQueryHandlerService,
+  EXPLORE_GQL_REQUEST,
+  GraphQlExploreResponse
+} from '../../../../graphql/request/handlers/explore/explore-graphql-query-handler.service';
+
+@Model({
+  type: 'trace-calls-percentage-data-source'
+})
+export class TraceCallsPercentageDataSourceModel extends GraphQlDataSourceModel<MetricAggregation> {
+  public getData(): Observable<MetricAggregation> {
+    return forkJoinSafeEmpty([this.fetchCallCountWithFilters(), this.fetchNumCallsData()]).pipe(
+      map(results => ({
+        value: getPercentage(results[0], results[1]),
+        health: MetricHealth.NotSpecified,
+        units: '%'
+      }))
+    );
+  }
+
+  private readonly context: string = 'API_TRACE';
```

Comment:
Can we also probably make this as ErrorPercentageDataSourceModel? Like this one can take context as a ModelProperty and call count key?

Reply:
sure. I didn't do that today because I couldn't find a usecase. thats all

Reply:
@skjindal93 done